docs/OCP_CI_Tutorials/Step_Registry/Step_Registry_Ref_Guide.md

```diff
@@ -0,0 +1,167 @@
+- `resources:` This stanza defines the resources that are needed to run your new ref.
+  - `requests:` This stanza defines the optimal resources needed to run this ref.
+    - `cpu:` The number of CPUs requested.
+    - `memory:` The amount of memory requested.
+  - `limits:` This stanza defines the maximum resources this ref should be allowed to utilize. **Optional**
+    - `cpu:` The number of CPUs requested.
+    - `memory:` The amount of memory requested.
+- `credentials:` This stanza defines a list of any credentials or secrets the ref may need for execution. Please see the [Secrets Guide](../Secrets/Secrets_Guide.md) for more information regarding the usage of this stanza. **Optional**
+  - `namespace:` Defines the credential's namespace defined in Vault.
+  - `name:` Defines the name of the credential in Vault.
+  - `mount_path:` Defines where the file(s) containing the credentials should be mounted during execution. If you have more than one item in this list, the `mount_path` values cannot be the same.
+- `env:` This stanza defines a list of environment variables needed for this ref to execute. **Optional**
+  - `name:` Defines the name of an environment variable.
+  - `default:` Defines the default value of the environment variable if one is not provided. **Optional if `env:` stanza is used**
+  - `documentation:` A short description of the environment variable and what it is used for.
+- `documentation:` The value of this should be a short description of the ref to be used in the [autogenerated documentation](../../Policy/Documentation/Step_Registry_Documentation_Policy.md#automatically-generated-documentation) in OpenShift CI.
+
+#### `new-step-commands.sh`
+This file is just the BASH script that will run during this ref's execution within the container you specified in the `from:` key of the ref's configuration file.
+
+> **IMPORTANT:**
+>
+> Just because this file is written in BASH, does not mean the tests or actions you are preforming have to be done in BASH. This file can be as simple as executing a Python script within the container. That being said, the `-commands` file does need to be a BASH script.
+
+Because this file is a BASH script, this document will not be diving in to the nuances of BASH. It will, however, give you some ideas of what can be accomplished in this script. Here are some examples:
+
+Utilizing the secrets defined in the `credentials` stanza of the ref's configuration file:
+```BASH
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+# Each key/value pair in a Vault secret is given a file in the mount_path defined. 
+# This files name is the key value and the contents is a single clear-text line containing the value.
+# See the Secrets Guide for more information.
+
+# This line is using the cat command to set the contents of a credential file to a variable.
+CREDENTIAL=$(cat /tmp/secrets/credentials/password)
```

Comment:
Have we verified that this is not outputting plain text passwords in the pod console output?

Reply:
Yes I have. It does not output anything in the MTR scenario. It is just setting the value of the variable to the output of the cat command, but it does not get printed.